src/main/java/org/folio/inventory/resources/MoveApi.java

```diff
@@ -103,41 +117,97 @@ private void updateItems(RoutingContext routingContext, WebContext context, List
         .map(CompletableFuture::join)
         .map(Item::getId)
         .collect(toList()))
-      .thenAccept(updatedItemIds -> {
-        List<String> nonUpdatedIds = ListUtils.subtract(itemIdsToUpdate, updatedItemIds);
-        HttpServerResponse response = routingContext.response();
-        if (nonUpdatedIds.isEmpty()) {
-          JsonResponse.successWithEmptyBody(response);
-        } else {
-          JsonResponse.successWithIds(response, nonUpdatedIds);
-        }
-      });
+      .thenAccept(updatedIds -> respond(routingContext, idsToUpdate, updatedIds));
   }
 
-  private OkapiHttpClient createHttpClient(RoutingContext routingContext, WebContext context) throws MalformedURLException {
-    return new OkapiHttpClient(client, context, exception -> ServerErrorResponse.internalError(routingContext.response(),
-        String.format("Failed to contact storage module: %s", exception.toString())));
+  private void moveHoldings(RoutingContext routingContext) {
+
+    WebContext context = new WebContext(routingContext);
+    JsonObject holdingsMoveJsonRequest = routingContext.getBodyAsJson();
+
+    Optional<ValidationError> validationError = holdingsMoveHasRequiredFields(holdingsMoveJsonRequest);
+    if (validationError.isPresent()) {
+      unprocessableEntity(routingContext.response(), validationError.get());
+    } else {
+      String toInstanceId = holdingsMoveJsonRequest.getString(TO_INSTANCE_ID);
+      List<String> holdingsRecordsIdsToUpdate = toListOfStrings(holdingsMoveJsonRequest.getJsonArray(HOLDINGS_RECORD_IDS));
```

Comment:
cane we have a separate class to represent a move holding request? Could deserialization be extracted to a separate method/or happens in this new class?

Reply:
I think no. We just need to extract only 2 fields from move schema, i.e. toInstanceId and holdingsRecordIds. And we haven't any further plans to extended this schema. So, I can propose to leave it as is. Please, confirm if you are OK with this.

Reply:
I'd prefer to have a class personally. 

It is a bit difficult to follow the code logic because of some additional nested blocks (ifs, catches, etc.), I left some comments, so I believe we can improve readability after resolving them.